src/lib/components/ProductForm.svelte

```diff
@@ -652,19 +691,39 @@
 							/>
 						</label>
 						<label class="form-label">
-							Value <input
+							{i === 0 ? 'Value' : ''}
+							<input
 								type="text"
 								name="variationLabels.values[{variation.name}][{variation.value}]"
 								class="form-input"
 								value={product.variationLabels?.values[variation.name]?.[variation.value]}
 								bind:this={variationInput[i]}
-								on:input={() => variationInput[i]?.setCustomValidity('')}
+								on:input={() => (
+									variationInput[i]?.setCustomValidity(''), (variationInput[i] = variationInput[i])
+								)}
 								required={!!product.variationLabels?.names[variation.name]}
 							/>
 						</label>
 					{:else}
+						<button
+							class="{i === 0 ? 'mt-8' : ''} {!variationLabelsNames[i] && !variationLabelsValues[i]
```

Comment:
Please bind the expression `!variationLabelsNames[i] && !variationLabelsValues[i]` to a variable after `{:else}` explaining what this represents. There are more instances of the same expression and it's really hard to understand.